### PR TITLE
feat(lean): add lean() in find query to improve performance

### LINKED ISSUE #53: lean
enable using .lean() to improve performance

http://www.tothenew.com/blog/high-performance-find-query-using-lean-in-mongoose-2/

### PR DESCRIPTION
**Pull request by mauriciomutte**
fix #53

### CODE REVIEW

**Comment by mauriciomutte** [comment]
`lean()` is not needed on aggregates, because the documents returned are plain JavaScript objects, and not Mongoose objects.

File: src/ConnectionFromMongoCursor.ts [hunk 1]
```diff
@@ -34,11 +34,12 @@ export const offsetToCursor = (offset: number): string => base64(PREFIX + offset
...
   cursor: Query<any>,
   useEstimatedCount?: boolean,
+  lean: boolean,
 };
 
-export const getTotalCount = async ({ cursor, useEstimatedCount = false }: TotalCountOptions): Promise<number> => {
+export const getTotalCount = async ({ cursor, useEstimatedCount = false, lean = true }: TotalCountOptions): Promise<number> => {
   // @ts-ignore
-  const clonedCursor = cursor.model.find().lean().merge(cursor);
+  const clonedCursor = lean ? cursor.model.find().lean().merge(cursor) : cursor.model.find().merge(cursor);
```
Comment (sibelius) [inline src/ConnectionFromMongoCursor.ts]: I think we can always use lean on totalCount

File: src/MongooseLoader.ts [hunk 2]
```diff
@@ -29,8 +29,9 @@ type Mongoose$Document = {
 export default async function mongooseLoader(
   model: Mongoose$Document,
   ids: ReadonlyArray<string>,
+  lean: true,
 ) {
-  const results = await model.find({ _id: { $in: ids } });
+  const results = await lean ? model.find({ _id: { $in: ids } }).lean() : model.find({ _id: { $in: ids } });
```
Comment (sibelius) [inline src/MongooseLoader.ts]: should await be `?`

**Comment by sibelius** [comment]
can you add tests using `lean` false? so we can make sure this is still working

**Pull request merged**